File: 30daysOFcode_Go/day 08.go
package main
import (
    "fmt"
    "bufio"
    "os"
)

func main() {
 //Enter your code here. Read input from STDIN. Print output to STDOUT
 var n int
 var k,v string
 m := make(map[string]string)
 scanner := bufio.NewScanner(os.Stdin)
 fmt.Scan(&n)
 for i := 0; i < n; i++ {
    fmt.Scanf("%s %s", &k, &v)
    m[k] = v
 }

 for scanner.Scan() {
    s := scanner.Text()
    if m[s] == "" {
        fmt.Println("Not found")
    } else {
        fmt.Printf("%s=%s\n",s,m[s])
    }
 }

}

File: 30daysOFcode_Scala/day 08.scala
object Solution {

  def main(args: Array[String]) {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution
*/
    var str :String=null
    val stdin = scala.io.StdIn
    val map = scala.collection.mutable.Map[String,String]()

    var t = stdin.readLine.trim.toInt
    while(t>0){
      str = stdin.readLine.trim
      val key=str.split(" ").array(0)
      val value=str.split(" ").array(1)
      map(key)=value
      t=t-1
    }
var flag=false


    while(flag==false){
        try{
            str = stdin.readLine.trim}catch{
        case x: Throwable => {flag=true}}
            if(str!=null && flag==false){
                try{println(str+"="+map(str))}
                catch{
                    case x => {println("Not found")}
                }
            }
        }
    }
}
